pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -509,11 +537,17 @@ func (nc NodeController) tryUpdateNodeStatus(node *api.Node) (error, time.Durati
 				nc.recordNodeOfflineEvent(node)
 			}
 		}
-		_, err = nc.kubeClient.Nodes().Update(node)
-		if err != nil {
-			glog.Errorf("Error updating node %s: %v", node.Name, err)
-		} else {
-			return nil, gracePeriod, lastReadyCondition
+		if !api.Semantic.DeepEqual(oldNodeStatus.status, node.Status) {
+			_, err = nc.kubeClient.Nodes().Update(node)
+			if err != nil {
+				glog.Errorf("Error updating node %s: %v", node.Name, err)
```

Comment:
I think its cleaner if you return here too (easier to understand the flow)

(while you're at it, it's weird that the error comes first in the return signature, mind fixing that while you're in here?)